### ISSUE TITLE
delete PNG files by default when rerunning SS_plots?

### ISSUE THREAD

**Issue by taylori** [r4ss/r4ss#150]
Currently, when you change a model and re-run SS_output and SS_plots, you get an error like this:

```
Error in SS_html(replist, filenotes = filenotes, plotdir = plotdir, ...,  : 
  CSV files with name 'plotInfoTable...' are from multiple model runs.
    Delete old files or (if you really know what you're doing) override with 'multimodel=TRUE.
    Runs:
     StartTime: Thu Mar 30 14:54:17 2017 
     StartTime: Thu Mar 30 13:38:54 2017 
```

It would be more convenient if this error were avoided by having the SS_plots function automatically delete the files in the "plots" folder when they are detected as being associated with an older model.

It would probably be safest to only delete PNG, CSV, and HTML files, which are the only types that r4ss creates, but leave the folder in place and not delete anything else that the user might have in there.

If anyone has a concern about this changes, post something here.

**Comment by kellijohnson** [comment]
I would prefer that the old plots be renamed based on the date-time that they were created and save the new plots with the default name. This is probably a pain to implement, but may save others in the long run.

**Comment by taylori** [comment]
Good idea, @kellijohnson, and thanks for the input.

I think renaming individual PNG files would be complex, but renaming the whole folder would be pretty easy. I can imagine a variety of ways this could be implemented, including
- copying the "plots" directory and its contents to something like "plots_2017-04-19_12.09.53" only if new plots are requested, with the new plots going into a new, empty "plots" directory.
- always using directory names like "plots_2017-04-19_12.09.53" by default.

In either case, the date/time should probably be based on when the model was run, not when the function was run.

The second option could make things harder for the template developed by @melmonk, but perhaps would have benefits that I'm not thinking of.

**Comment by taylori** [comment]
Also, note that all this can be avoided if you just run each iteration of the model in a new directory. However, in practice, there are lots of model runs that are obviously bad and not worth archiving and it's quicker to just rerun the next iteration in the same place.

**Comment by allanhicks** [comment]
If you simply rename the plots directory to "plots_2017-04-19_12.09.53", it
will save a lot of time due to copying. And, then let the code run as
normal, which creates a pots directory anyways (when one is not present).

I like the idea of maintaining the "plots" directory because it maintains
backwards compatibility and is generic for code writing and automation.

**Comment by taylori** [comment]
This should be sorted in out commit 9fa4ce4. As discussed, the SS_plots function will rename "plots" to something following the format "plots_StartTime_Th_Ap_2_13.18.3_2017" whenever the CSV files containing information on the plots indicates that they are from an earlier model run than the one passed to SS_plots. If there's no conflict, the default directory name will remain "plots".

I'll wait until at least one other person tests this and reports that it works before closing the issue.

**Comment by kellijohnson** [comment]
maybe @CWetzel can test it because she is using the markdown template?

**Comment by CWetzel** [comment]
Tested, but I don't think it is quite there.  The 'plot' folder was not renamed.  Within the plot folder the html files were overwritten, did not delete or overwrite the png figures, but did create time stamped csv file within the plot folder.

**Comment by taylori** [comment]
Thanks for testing @CWetzel.
Did you re-run the model? I set it so that if the model output hasn't changed (as indicated by StartTime in Report.sso), then it will just overwrite the plots. 

It looks like the @melmonk markdown template has you rename "plots" to "plots_mod1", so there's little danger of accidentally getting the wrong set.

**Comment by CWetzel** [comment]
Alright, I reran the model and it did rename my older plot folder and create a new plot folder with png and html files.  However, the rename of the plot folder was "plots_StartTime_Th_Ap_2_10.27.1_2017" when I think it should be **plots_StartTime_Th_Ap_2**0**_10.27.1_2017".  

This update is not an issue for the template Melissa Monk created because she deletes all files within "plot_mod1" folder first every time a model is loaded.  Also, she sets the "dir" input to be the specifically named plot folder (plot_mod1) rather than allowing the folder to take the default name of plot

**Comment by taylori** [comment]
Thanks @CWetzel. I was sloppy in my substituting the ":" and blanks in the path. I thought "Th" and "Ap" were strange but assumed it was just the ADMB way of writing the day and month. I failed to notice that 20 became 2, which is a bigger issue.  
Should now be fixed in e4441f5c09b6b73350e69b14878238704561ad33 so will close. We can reopen if that didn't really fix it.

**Comment by CWetzel** [comment]
It now gives the correct date.  Thank you.
